[regression] WriteCodeFragment fails if parameter includes brackets

**Original customer report:** https://github.com/aspnet/Configuration/issues/555

WriteCodeFragment fails if the parameter includes matching brackets.

**Repro**
```xml
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="Build">
        <ItemGroup>
            <Attr Include="TestAttribute">
                <_Parameter1>Some {value}</_Parameter1>
            </Attr>
        </ItemGroup>
        <WriteCodeFragment Language="C#" AssemblyAttributes="@(Attr)" OutputFile="attr.cs" />
    </Target>
</Project>
```

**Info:**
Using dotnet-CLI 1.0.0-preview3-004056 on Windows 10.
Doesn't repo MSBuild 14.0.

**Stack trace:**
```
  "C:\tmp\test.proj" (default target) (1) ->
       (Build target) ->
         C:\tmp\test.proj(8,9): error MSB4018: The "WriteCodeFragment" task failed unexpectedly.\r
       C:\tmp\test.proj(8,9): error MSB4018: System.FormatException: Input string was not in a correct format.\r
       C:\tmp\test.proj(8,9): error MSB4018:    at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)\r
       C:\tmp\test.proj(8,9): error MSB4018:    at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)\r
       C:\tmp\test.proj(8,9): error MSB4018:    at System.String.Format(String format, Object[] args)\r
       C:\tmp\test.proj(8,9): error MSB4018:    at Microsoft.Build.Tasks.WriteCodeFragment.GenerateCodeCoreClr(String& extension)\r
       C:\tmp\test.proj(8,9): error MSB4018:    at Microsoft.Build.Tasks.WriteCodeFragment.Execute()\r
       C:\tmp\test.proj(8,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()\r
       C:\tmp\test.proj(8,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext()
```

--- a/src/Tasks.UnitTests/WriteCodeFragment_Tests.cs
+++ b/src/Tasks.UnitTests/WriteCodeFragment_Tests.cs
@@ -426,6 +426,30 @@ namespace Microsoft.Build.UnitTests
             File.Delete(task.OutputFile.ItemSpec);
         }
 
+        [Fact]
+        public void OneAttributeTwoSamePositionalParams()
+        {
+            WriteCodeFragment task = new WriteCodeFragment();
+            MockEngine engine = new MockEngine(true);
+            task.BuildEngine = engine;
+            TaskItem attribute = new TaskItem("AssemblyMetadataAttribute");
+            attribute.SetMetadata("_Parameter1", "TestValue");
+            attribute.SetMetadata("_Parameter2", "TestValue");
+            task.AssemblyAttributes = new TaskItem[] { attribute };
+            task.Language = "c#";
+            task.OutputDirectory = new TaskItem(Path.GetTempPath());
+            bool result = task.Execute();
+
+            Assert.Equal(true, result);
+
+            string content = File.ReadAllText(task.OutputFile.ItemSpec);
+            Console.WriteLine(content);
+
+            CheckContentCSharp(content, @"[assembly: AssemblyMetadataAttribute(""TestValue"", ""TestValue"")]");
+
+            File.Delete(task.OutputFile.ItemSpec);
+        }
+
         public static string EscapedLineSeparator => NativeMethodsShared.IsWindows ? "\\r\\n" : "\\n";
 
         /// <summary>

--- a/src/Tasks.UnitTests/WriteCodeFragment_Tests.cs
+++ b/src/Tasks.UnitTests/WriteCodeFragment_Tests.cs
@@ -427,7 +427,7 @@ namespace Microsoft.Build.UnitTests
         }
 
         [Fact]
-        public void OneAttributeTwoSamePositionalParams()
+        public void OneAttributeTwoPositionalParamsWithSameValue()
         {
             WriteCodeFragment task = new WriteCodeFragment();
             MockEngine engine = new MockEngine(true);

--- a/src/Tasks/WriteCodeFragment.cs
+++ b/src/Tasks/WriteCodeFragment.cs
@@ -384,7 +384,7 @@ namespace Microsoft.Build.Tasks
                 }
             }
 
-            return string.Join(", ", orderedParameters.Union(namedParameters).Where(p => !string.IsNullOrWhiteSpace(p)));
+            return string.Join(", ", orderedParameters.Concat(namedParameters).Where(p => !string.IsNullOrWhiteSpace(p)));
         }
 
         private const int MaxLineLength = 80;

